### PR TITLE
Adds Ethereum signatures verification + refactor the orchestrator/relayer for testing

### PR DESCRIPTION
**Pull request by SweeXordious**

### CODE REVIEW

**Comment by SweeXordious** [comment]
@evan-forbes I think this PR can be merged so that the upcoming changes can be tracked in separate PRs.

**Review by evan-forbes — APPROVED** [review]
cool, nice additions, and sets us up for future changes

File: go.sum [hunk 1]
```diff
@@ -1,4 +1,7 @@
+4d63.com/gochecknoglobals v0.0.0-20201008074935-acfc0b28355a/go.mod h1:wfdC5ZjKSPr7CybKEcgJhUOgeAQW1+7WcyK8OvUilfo=
```
Comment (adlerjohn) [inline go.sum:1]: why is `go.sum` being modified?

File: x/qgb/abci.go [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
 // EndBlocker is called at the end of every block
 func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
 	// TODO cleanup
-	fmt.Println("running the qgb endblocker ======================================================")
-	// get the last valsets to compare against
+	//get the last valsets to compare against
```
Comment (adlerjohn) [inline x/qgb/abci.go:13]: Is this still a todo?

File: go.sum (hunk 1 above)
Comment (SweeXordious) [reply go.sum:1]: because we keep adding new dependencies etc, I guess

File: x/qgb/abci.go (hunk 2 above)
Comment (SweeXordious) [reply x/qgb/abci.go:13]: will remove it. thanks

**Pull request merged**